src/rest-api/endpoints/management-endpoint/search-endpoint.md

```diff
@@ -0,0 +1,248 @@
+| **nxql**       | **string** | The NXQL query.                     | Optional, no query means all visible documents  |
+| **pageSize**   | **number** | The number of documents to return.  | Optional, default is 10                         |
+
+### Response
+
+If successful, it will return a JSON response containing search results from all available search indexes for the repository.
+
+For instance, `repository/repository` is the response from the repository Search Client. If you have installed the `nuxeo-search-client-opensearch1` package  the search response from the OpenSearch index is under `opensearch/enhanced`.
+
+Note that only document identifiers are returned, this is on purpose because management endpoint should not expose document content.
+
+### Status Codes
+
+- 200 *OK* - Success.
+
+### Sample
+
+To check the number of visible documents in OpenSearch and in the repository:
+
+```curl
+curl -X GET -u Administrator:Administrator \
+   --data-urlencode "nxql=SELECT * FROM Document WHERE ecm:isProxy = 0 AND ecm:isVersion = 0 AND ecm:isTrashed = 0" \
+   --data-urlencode "pageSize=5" \
+   http://localhost:8080/nuxeo/api/v1/management/search/checkSearch
+```
+
+```json
+{
+ "query": "SELECT * FROM Document WHERE ecm:mixinType != 'HiddenInNavigation' AND ecm:isProxy = 0 AND ecm:isVersion = 0 AND ecm:isTrashed = 0",
+  "order": {
+    "sortColumn": "dc:modified",
+    "sortAscending": false
+  },
+  "repository/repository": {
+    "took": 3,
+    "resultsCount": 17,
+    "pageSize": 10,
+    "pageProvider": "search_check_nxql",
+    "resultsCountLimit": 10000,
+    "results": [
+      "e58dbf5c-fca7-4986-a664-44575aef01ac",
+      "d057e3b6-2a96-4a18-a49e-c061c85eebaa",
+      "5af22aaf-5dfd-4d64-b001-d46bd8777e02",
+      "7d2c2aec-f414-483e-add7-d636c5887b1e",
+      "689c4560-4873-41c2-8b5f-7a6a92d831b0",
+      "59a68ac8-f7cd-4462-802d-5d460cd5343b",
+      "674ee490-3918-443b-bca7-b2fde1127085",
+      "4166773f-4584-48b4-9e75-2421d7ab8e47",
+      "ab36fd80-c4f3-4313-86e2-35f1d196742c",
+      "58ae218e-99ae-4ac7-b25b-4517759e4402"
+    ]
+  },
+  "opensearch/enhanced": {
+    "took": 15,
+    "resultsCount": 17,
+    "pageSize": 10,
+    "pageProvider": "search_check_nxql",
+    "resultsCountLimit": 10000,
+    "results": [
+      "e58dbf5c-fca7-4986-a664-44575aef01ac",
+      "d057e3b6-2a96-4a18-a49e-c061c85eebaa",
+      "5af22aaf-5dfd-4d64-b001-d46bd8777e02",
+      "7d2c2aec-f414-483e-add7-d636c5887b1e",
+      "689c4560-4873-41c2-8b5f-7a6a92d831b0",
+      "59a68ac8-f7cd-4462-802d-5d460cd5343b",
+      "674ee490-3918-443b-bca7-b2fde1127085",
+      "4166773f-4584-48b4-9e75-2421d7ab8e47",
+      "ab36fd80-c4f3-4313-86e2-35f1d196742c",
+      "58ae218e-99ae-4ac7-b25b-4517759e4402"
+    ]
+  }
+}
+```
+
+If there are discrepancies, further investigation can be done using directly the page providers (`nxql_repo_search` or `nxql_elastic_search`).
```

Comment:
The two mentioned PPs are not present in the 2025 code base.

Reply:
Good catch, atm there is no way to investigate with curl.
We need to have a way to force SearchService PP to use an index (like a header is used to set the repository) I will create an NXP for that, doc will be updated again, after.